src/main/java/org/openrewrite/kotlin/internal/KotlinTreeParserVisitor.java

```diff
@@ -3259,6 +3259,8 @@ private K.Binary.Type mapKBinaryType(KtOperationReferenceExpression operationRef
             return K.Binary.Type.IdentityNotEquals;
         } else if (elementType == KtTokens.ELVIS) {
             return K.Binary.Type.Elvis;
+        } else if (elementType == KtTokens.PERCEQ) {
+            return K.Binary.Type.ModAssign;
```

Comment:
This should be a `J.AssignmentOperation` or similar. In Java we also have `%=`: https://github.com/openrewrite/rewrite/blob/067c32ed93aa873b27d233ca6cf6f7608da1db99/rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java#L578-L581

Reply:
good call, thanks, revised it to use `J.AssignmentOperation` instead of Binary